### PR TITLE
DocumentCard, Pivot, GroupHeader: Remove use of neutralSecondaryAlt

### LINKED ISSUE #6500: neutralSecondaryAlt color needs to be replaced across the repo
### Bug Report 

- __Package version(s)__: all
 
#### Priorities and help requested:
 
Are you willing to submit a PR to fix? Yes as long as we have replacement colors from designers
 
Requested priority: (High) 

 
#### Describe the issue:
 
neutralSecondaryAlt was deprecated several months ago, we should remove any instances of it.
First step of removing it was in #6499 as it had a real impact on the themed SPList.
Most likely will be needed designers to provide a replacement color for where it's used.

### PR DESCRIPTION
**Pull request by natalieethell**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #6500 
- [X] Include a change request file using `$ npm run change`

#### Description of changes

These changes replace instances of `neutralSecondaryAlt` with the appropriate colors. DocumentCard and GroupHeader now use `neutralSecondary` in those places. Pivot's outline now uses `semanticColors.focusBorder`. Additionally, as discussed with Ben, `focusBorder` should be `neutralSecondary`, not black. These changes are reflected here as well. 

I also added a new test for DocumentCardTitle with its `showAsSecondaryTitle` prop set to true.

#### Focus areas to test

* In the new DocumentCardTitle test with `showAsSecondaryTitle` prop set to true, the color attribute of `ms-DocumentCardTitle` should be `neutralSecondary`.
* In Pivot, the focus outline should be `neutralSecondary`.
* In GroupHeader, the drop icon should be `neutralSecondary`.

*** A valid concern @Vitalius1 brought up ***
If the old `focusBorder` was used in the consuming apps of Fabric, will this break their expectation on design? Thoughts?

### CODE REVIEW

**Comment by natalieethell** [comment]
Note: Still need to update snapshot tests again with the corresponding change to `focusBorder`.

File: packages/office-ui-fabric-react/src/components/DocumentCard/__snapshots__/DocumentCardTitle.test.tsx.snap [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`DocumentCardTitle shows title with secondary title style 1`] = `
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/DocumentCard/__snapshots__/DocumentCardTitle.test.tsx.snap]: Not sure if this snapshot is seeing the difference in between showAsSecondaryTitle set to true or false.
If the intention was to check for color difference than maybe a VR story would be more appropriate.

Comment (natalieethell) [reply packages/office-ui-fabric-react/src/components/DocumentCard/__snapshots__/DocumentCardTitle.test.tsx.snap]: You're totally right, good call. I'll write a VR test instead.

**Comment by natalieethell** [comment]
Hey @betrue-final-final I added a new visual regression test, could you take a look when you have a chance?

**Pull request merged**